godot-macros/src/docs.rs

```diff
@@ -135,13 +151,76 @@ fn xml_escape(value: String) -> String {
     result
 }
 
-/// Calls [`extract_docs_from_attributes`] and converts the result to BBCode for Godot's consumption.
-fn attribute_docs_to_bbcode(doc: &[venial::Attribute]) -> Option<String> {
-    let doc = extract_docs_from_attributes(doc)
-        .collect::<Vec<String>>()
-        .join("\n");
+/// Extracts docs from attributes and groups them in three Strings:
+/// user documentation content and paragraphs annotated with `@deprecated` or `@experimental` tags.
+fn docs_with_attributes(doc: &[venial::Attribute]) -> (String, String, String) {
+    let (mut docs, mut deprecated, mut experimental) =
+        (String::new(), String::new(), String::new());
+
+    // Allows to compare the current bucket (the one we put current paragraph in) with docs one.
+    let docs_bucket = std::ptr::from_ref(&docs);
+    let mut current_bucket: &mut String = &mut docs;
+
+    for line in extract_docs_from_attributes(doc) {
+        let trimmed = line.trim_start();
+
+        // End of the paragraph (`#[doc=""]` or `///`) .
+        if trimmed.is_empty() {
+            // Switch back from attribute docs to user docs when paragraph ends.
+            // Don't double newlines after XML attribute tags descriptions.
+            if !std::ptr::eq(current_bucket, docs_bucket) {
+                current_bucket = &mut docs;
+            } else {
+                // End of the paragraph.
```

Comment:
I'll remove this comment (doubles with the one above) during the squashing or while pushing other fixes.